Sources/Articles/Documentation.docc/2024/beginners-guide-to-protocol-buffers-and-grpc-with-swift/beginners-guide-to-protocol-buffers-and-grpc-with-swift.md

```diff
@@ -0,0 +1,360 @@
+# Beginner's Guide to Protocol Buffers and gRPC with Swift
+
+This tutorial will explore Protocol Buffers and the gRPC communication service. For those unfamiliar with these technologies, this guide provides an excellent introduction to using them with Swift.
+
+## What Are Protocol Buffers?
+
+Protocol Buffers [protobuf](https://protobuf.dev/) is a data serialization method developed by Google. It is particularly useful for enabling programs to communicate over a network or for efficiently storing structured data.
+
+Protocol Buffers are extensively used across various applications, such as network communication, data storage, and configuration management. They offer a language-neutral, platform-neutral, and extensible approach to serializing structured data.
+
+## What is gRPC?
+
+[gRPC](https://grpc.io/) (gRPC Remote Procedure Call) is an open-source framework developed by Google. It enables efficient communication between distributed systems and services, allowing them to collaborate seamlessly.
+
+It is quite similar to the OpenAPI standard, but Protocol Buffers are more efficient, as they use a binary format instead of JSON. 
+
+When using the OpenAPI format, schemas are defined using a JSON or YAML file. In contrast, Protocol Buffers define the schema of data structures using a `.proto` file, which specifies the structure and data types to be serialized.
+
+
+## The Protocol Buffer Compiler
+
+The protocol buffer compiler (protoc) is a tool that generates source code for data serialization and gRPC communication from `.proto` schema files.
+
+The protocol buffers compiler supports many [languages](https://protobuf.dev/getting-started/), including Swift. To [install](https://grpc.io/docs/protoc-installation/) protoc on your machine, follow the instructions based on your operating system:
+
+```sh
+# install on macOS 
+brew install protobuf
+
+# install on Linux
+apt install -y protobuf-compiler
+
+# manual install, on any operating system
+PB_REL="https://github.com/protocolbuffers/protobuf/releases"
+curl -LO $PB_REL/download/v25.1/protoc-25.1-linux-x86_64.zip
+unzip protoc-25.1-linux-x86_64.zip -d $HOME/.local
+export PATH="$PATH:$HOME/.local/bin"
```

Comment:
I propose writing about the language before the compiler. So people can get their hands dirty first.
Also consider mentioning a VSCode plugin for protobuf.